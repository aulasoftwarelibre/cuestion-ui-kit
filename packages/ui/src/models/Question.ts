export interface Question {
  id: string;
  question: string;
  createdAt: Date;
  username: string;
  votes: number;
  isVoted: boolean;
}

export default Question;
