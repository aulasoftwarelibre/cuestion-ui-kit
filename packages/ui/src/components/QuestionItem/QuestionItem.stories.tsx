import { action } from "@storybook/addon-actions";
import { object, withKnobs } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";

import { createQuestion, question } from "../../models/Question.mock";
import { wInfo } from "../../utils/wInfo";
import QuestionItem from "./QuestionItem";

const stories = storiesOf("Components/QuestionItem", module);
stories.addDecorator(withKnobs);

stories
  .add(
    "Not Liked Question",
    wInfo()(() => (
      <QuestionItem
        handleOnClick={action("question")}
        question={object("Question", createQuestion("1", 10, false))}
      />
    )),
  )
  .add(
    "Liked Question",
    wInfo()(() => (
      <QuestionItem
        handleOnClick={action("question")}
        question={object("Question", createQuestion("1", 10, true))}
      />
    )),
  );
