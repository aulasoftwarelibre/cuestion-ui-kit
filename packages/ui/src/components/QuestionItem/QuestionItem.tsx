import {
  Avatar,
  Card,
  CardActions,
  CardContent,
  CardHeader,
  IconButton,
  Typography,
} from "@material-ui/core";
import { red } from "@material-ui/core/colors";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import FavoriteIcon from "@material-ui/icons/Favorite";
import FavoriteOutlinedIcon from "@material-ui/icons/FavoriteBorder";

import * as React from "react";
import { InjectedIntl, injectIntl } from "react-intl";
import { Question } from "../../models/Question";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    card: {
      minWidth: 320,
    },
    media: {
      height: 0,
      paddingTop: "56.25%", // 16:9
    },
    expand: {
      transform: "rotate(0deg)",
      marginLeft: "auto",
      transition: theme.transitions.create("transform", {
        duration: theme.transitions.duration.shortest,
      }),
    },
    expandOpen: {
      transform: "rotate(180deg)",
    },
    avatar: {
      backgroundColor: red[500],
    },
  }),
);

interface Props {
  question: Question;
  handleOnClick: any;
  intl: InjectedIntl;
}

const _QuestionItem: React.FunctionComponent<Props> = ({
  handleOnClick,
  question,
  intl,
}) => {
  const classes = useStyles();

  const creationHour = intl.formatTime(question.createdAt);

  return (
    <Card className={classes.card}>
      <CardHeader
        avatar={
          <Avatar aria-label="User avatar" className={classes.avatar}>
            {question.username.charAt(0)}
          </Avatar>
        }
        title={question.username}
        subheader={creationHour}
      />
      <CardContent>
        <Typography variant="h6" color="primary" component="p">
          {question.question}
        </Typography>
      </CardContent>
      <CardActions disableSpacing>
        <IconButton
          aria-label="Like button"
          onClick={() => handleOnClick(question.id, !question.isVoted)}>
          {question.isVoted ? <FavoriteIcon /> : <FavoriteOutlinedIcon />}
        </IconButton>
        <Typography variant="h6" color="primary" component="p">
          {question.votes}
        </Typography>
      </CardActions>
    </Card>
  );
};

export const QuestionItem = injectIntl(_QuestionItem);

export default QuestionItem;
