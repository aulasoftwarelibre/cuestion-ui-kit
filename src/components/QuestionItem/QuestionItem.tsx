import { Avatar, Card, CardActions, CardContent, CardHeader, IconButton, Typography } from "@material-ui/core";
import { red } from "@material-ui/core/colors";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import FavoriteIcon from "@material-ui/icons/Favorite";

import * as React from "react";
// import { FormattedMessage, InjectedIntl, injectIntl } from "react-intl";
import { Question } from "../../models/Question";

// import messages from "../../languages/messages";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    card: {
      width: 100
    },
    media: {
      height: 0,
      paddingTop: "56.25%" // 16:9
    },
    expand: {
      transform: "rotate(0deg)",
      marginLeft: "auto",
      transition: theme.transitions.create("transform", {
        duration: theme.transitions.duration.shortest
      })
    },
    expandOpen: {
      transform: "rotate(180deg)"
    },
    avatar: {
      backgroundColor: red[500]
    }
  })
);

interface Props {
  question: Question;
  handleOnClick: any;
  // intl: InjectedIntl;
}

const _QuestionItem: React.FunctionComponent<Props> = ({ handleOnClick, question }) => {
  const [pressed, setPressed] = React.useState<boolean>(!question.isVoted);

  const classes = useStyles();

  const handleLikeButton = () => {
    handleOnClick(question.id);

    setPressed(!pressed);
    question.isVoted = pressed;

    if (question.isVoted) {
      question.votes = question.votes + 1;
    } else {
      question.votes = question.votes - 1;
    }
  };

  return (
    <Card>
      <CardHeader
        avatar={
          <Avatar aria-label="User avatar" className={classes.avatar}>
            X
          </Avatar>
        }
        title={question.username}
        subheader={question.createdAt.toUTCString()}
      />
      <CardContent>
        <Typography variant="body2" color="textSecondary" component="p">
          {question.question}
        </Typography>
      </CardContent>
      <CardActions disableSpacing>
        <IconButton aria-label="Like button" onClick={() => handleLikeButton()}>
          <FavoriteIcon />
        </IconButton>
        {question.votes}
      </CardActions>
    </Card>
  );
};

// export const QuestionItem = injectIntl(_QuestionItem);

export default _QuestionItem;
