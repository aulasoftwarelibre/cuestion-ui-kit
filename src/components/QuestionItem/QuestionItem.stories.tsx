import { action } from "@storybook/addon-actions";
import { object, withKnobs } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";

import { question, question2 } from "../../models/Question.mock";

import { wInfo } from "../../utils/wInfo";
import QuestionItem from "./QuestionItem";

const stories = storiesOf("Components/QuestionItem", module);
stories.addDecorator(withKnobs);

stories
  .add(
    "Not Liked Question",
    wInfo()(() => <QuestionItem handleOnClick={action("question")} question={object("Question", question)} />)
  )
  .add(
    "Liked Question",
    wInfo()(() => <QuestionItem handleOnClick={action("question")} question={object("Question", question2)} />)
  );
