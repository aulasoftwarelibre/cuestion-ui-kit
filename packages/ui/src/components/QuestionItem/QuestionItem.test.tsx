import Chip from "@material-ui/core/Chip";
import * as React from "react";

import { question, question2 } from "../../models/Question.mock";
import {
  createComponentWithIntl,
  mountWithIntl,
} from "../../utils/createComponentWithIntl";
import QuestionItem from "./QuestionItem";

describe("SessionInput", () => {
  const mockOnHandleClick = jest.fn();

  it("should render not liked question snapshot", () => {
    const component = createComponentWithIntl(
      <QuestionItem handleOnClick={mockOnHandleClick} question={question} />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it("should render liked question snapshot", () => {
    const component = createComponentWithIntl(
      <QuestionItem handleOnClick={mockOnHandleClick} question={question2} />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it("should return isVoted is true", () => {
    const component = mountWithIntl(
      <QuestionItem handleOnClick={mockOnHandleClick} question={question} />,
    );

    component.find(Chip).simulate("click");

    expect(mockOnHandleClick).toHaveBeenCalledWith(
      question.id,
      !question.isVoted,
    );
  });

  it("should return isVoted is false", () => {
    const component = mountWithIntl(
      <QuestionItem handleOnClick={mockOnHandleClick} question={question2} />,
    );

    component.find(Chip).simulate("click");

    expect(mockOnHandleClick).toHaveBeenCalledWith(
      question2.id,
      !question2.isVoted,
    );
  });
});
