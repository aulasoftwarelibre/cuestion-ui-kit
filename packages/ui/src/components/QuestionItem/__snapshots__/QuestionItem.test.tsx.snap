// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionInput should render liked question snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-avatar"
    >
      <div
        aria-label="User avatar"
        className="MuiAvatar-root makeStyles-avatar-5 MuiAvatar-colorDefault"
      >
        M
      </div>
    </div>
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
      >
        Miguel Angel
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        9:30 AM
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary"
    >
      ¿Cuantos años tienes?
    </p>
  </div>
  <div
    className="MuiCardActions-root"
  >
    <button
      aria-label="Like button"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
          />
        </svg>
      </span>
    </button>
    <p
      className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary"
    >
      10
    </p>
  </div>
</div>
`;

exports[`SessionInput should render not liked question snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-avatar"
    >
      <div
        aria-label="User avatar"
        className="MuiAvatar-root makeStyles-avatar-5 MuiAvatar-colorDefault"
      >
        J
      </div>
    </div>
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
      >
        Javier Velasco
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        9:30 AM
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary"
    >
      ¿Como te llamas?
    </p>
  </div>
  <div
    className="MuiCardActions-root"
  >
    <button
      aria-label="Like button"
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
          />
        </svg>
      </span>
    </button>
    <p
      className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary"
    >
      0
    </p>
  </div>
</div>
`;
